src/tasks/index.ts

```diff
@@ -0,0 +1,174 @@
+	if (db.engine !== 'postgres') {
+		return;
+	}
+
+	// Create trigger function if it doesn't exist
+	await createTrigger(tx);
+
+	// Create indexes if they don't exist
+	await createIndexes(tx);
+
+	const client = new PinejsClient({
+		apiPrefix: `/${apiRoot}/`,
+	});
+	worker = new Worker(client);
+
+	// Add resource hooks
+	addPureHook('POST', apiRoot, 'task', {
+		POSTPARSE: async ({ req, request }) => {
+			// Set the actor
+			request.values.is_created_by__actor =
+				req.user?.actor ?? req.apiKey?.actor;
+			if (request.values.is_created_by__actor == null) {
+				throw new BadRequestError(
+					'Creating tasks with missing actor on req is not allowed',
+				);
+			}
+
+			// Set defaults
+			request.values.status = 'queued';
+			request.values.attempt_count = 0;
+			request.values.attempt_limit ??= 1;
+			// TODO: Implement a balancer to better enqueue tasks based on actor usage
+			request.values.priority ??= 1;
+
+			// Set scheduled start time using cron expression if provided
+			if (
+				request.values.is_scheduled_with__cron_expression != null &&
+				request.values.is_scheduled_to_execute_on__time == null
+			) {
+				try {
+					request.values.is_scheduled_to_execute_on__time = cronParser
+						.parseExpression(request.values.is_scheduled_with__cron_expression)
+						.next()
+						.toDate()
+						.toISOString();
+				} catch (_) {
+					throw new BadRequestError(
+						`Invalid cron expression: ${request.values.is_scheduled_with__cron_expression}`,
+					);
+				}
+			}
+
+			// Assert that the provided start time is far enough in the future
+			if (request.values.is_scheduled_to_execute_on__time != null) {
+				const now = new Date(new Date().getTime() + tasksEnv.queueIntervalMS);
+				const startTime = new Date(
+					request.values.is_scheduled_to_execute_on__time,
+				);
+				if (startTime < now) {
+					throw new BadRequestError(
+						`Task scheduled start time must be greater than ${tasksEnv.queueIntervalMS} milliseconds in the future`,
+					);
+				}
+			}
+
+			// Assert that the requested handler exists
+			const handlerName = request.values.is_executed_by__handler;
+			if (handlerName == null) {
+				throw new BadRequestError(`Must specify a task handler to execute`);
+			}
+			const handler = worker?.handlers[handlerName];
+			if (handler == null) {
+				throw new BadRequestError(
+					`No task handler with name '${handlerName}' registered`,
+				);
+			}
+
+			// Assert that the provided parameter set is valid
+			if (handler.validate != null) {
+				if (!handler.validate(request.values.is_executed_with__parameter_set)) {
+					throw new BadRequestError(
+						`Invalid parameter set: ${ajv.errorsText(handler.validate.errors)}`,
+					);
+				}
+			}
+		},
+	});
+	worker.start();
+}
+
+// Register a task handler
+export function addTaskHandler(
+	name: string,
+	fn: TaskHandler['fn'],
+	schema?: Schema,
+): void {
+	if (worker == null) {
+		return;
+	}
```

Comment:
I assume this is because we only support tasks on postgres, which is fine but should throw an error here rather than silently/surprisingly doing nothing